scripts/w35-cache-trace.sh

```
@@ -0,0 +1,544 @@
+
+# Seed Claude Code's onboarding flag in a fresh HOME so headless mode never
+# stalls on first-run setup.
+seed_home() { mkdir -p "$1"; printf '{"hasCompletedOnboarding": true}\n' > "$1/.claude.json"; }
+
+# Run ONE headless session whose stdout is captured to $log. Output format is
+# stream-json --verbose so the per-turn + result usage records (with the cache
+# token breakdown) are captured. Cheap model + hard spend cap + wall-clock cap.
+run_session() {
+  local home="$1" proj="$2" prompt="$3" log="$4"; shift 4
+  (
+    export HOME="$home"
```

Shellcheck (0.11.0)

[info] Modification of HOME is local (to subshell caused by (..) group).

(SC2030)